dissect/target/plugins/general/osinfo.py

```diff
@@ -0,0 +1,36 @@
+from typing import Callable, Generator
+
+from flow.record import Record
+
+from dissect.target import plugin
+from dissect.target.exceptions import PluginError
+from dissect.target.helpers.record import TargetRecordDescriptor
+
+OSInfoRecord = TargetRecordDescriptor(
+    "generic/osinfo",
+    [
+        ("string", "prop"),
+        ("string", "value"),
+    ],
+)
+
+
+class OSInfoPlugin(plugin.Plugin):
+    """Convenience plugin that wraps _os.* functions in records."""
+
+    def check_compatible(self) -> bool:
+        return True
+
+    @plugin.export(record=OSInfoRecord)
+    def osinfo(self) -> Generator[Record, None, None]:
+        for os_func in self.target._os.__functions__:
+            value = getattr(self.target._os, os_func)
+            # Aggregate functions that yield records of their own
+            if isinstance(value, Callable):
+                try:
+                    value = len(list(value()))
+                except (PluginError, TypeError):
+                    # Ignore exceptions triggered by functions
+                    # that cannot be executed in this context
+                    continue
+            yield OSInfoRecord(prop=os_func, value=value)
```

Comment:
```suggestion
            yield OSInfoRecord(prop=os_func, value=value, _target=self.target)
```
Or was that deliberate?